dbms/src/Operators/ExpressionTransformOp.cpp

```diff
@@ -17,15 +17,31 @@
 
 namespace DB
 {
-OperatorStatus ExpressionTransformOp::transformImpl(Block & block)
+template <bool selective>
+OperatorStatus ExpressionTransformOp<selective>::transformImpl(Block & block)
 {
     if (likely(block))
-        expression->execute(block);
+    {
+        if constexpr (selective)
```

Comment:
It seems not necessary to use template here since compare to `expression->execute(block)`, the extra cost of the if branch can be ignored.

Reply:
ok